mcp/mcp-cli/src/main/java/software/amazon/smithy/java/mcp/cli/commands/ClientConfigCommand.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.mcp.cli.commands;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+import software.amazon.smithy.java.mcp.cli.ConfigUtils;
+import software.amazon.smithy.java.mcp.cli.ConfigurationCommand;
+import software.amazon.smithy.java.mcp.cli.ExecutionContext;
+import software.amazon.smithy.java.mcp.cli.SmithyMcpCommand;
+import software.amazon.smithy.java.mcp.cli.model.ClientConfig;
+
+@Command(
+        name = "client-config",
+        description = "Configure MCP client settings",
+        subcommands = {
+                ClientConfigCommand.ListCommand.class,
+                ClientConfigCommand.AddCommand.class,
+                ClientConfigCommand.RemoveCommand.class
+        })
+final class ClientConfigCommand {
+
+    @Command(name = "list", description = "List all configured MCP clients")
+    public static final class ListCommand extends SmithyMcpCommand implements ConfigurationCommand {
+
+        @Override
+        protected void execute(ExecutionContext context) throws Exception {
+            var clientConfigs = context.config().getClientConfigs();
+
+            if (clientConfigs == null || clientConfigs.isEmpty()) {
+                System.out.println("No client configurations found.");
+                return;
+            }
+
+            System.out.println("Configured MCP clients:");
+            var userHome = System.getProperty("user.home");
+            for (var clientConfig : clientConfigs) {
+                if (Boolean.TRUE.equals(clientConfig.isDisabled())) {
+                    continue;
+                }
+                var defaultMarker = clientConfig.isIsDefault() ? " (default)" : "";
+                var displayPath = clientConfig.getFilePath();
+                if (displayPath.startsWith(userHome)) {
+                    displayPath = "~" + displayPath.substring(userHome.length());
+                }
+                System.out.printf("  - %s: %s%s%n",
+                        clientConfig.getName(),
+                        displayPath,
+                        defaultMarker);
+            }
+        }
+    }
+
+    @Command(name = "add", description = "Add a new MCP client configuration")
+    public static final class AddCommand extends SmithyMcpCommand implements ConfigurationCommand {
+
+        @Parameters(index = "0", description = "Client name (e.g., 'q-cli')")
+        String clientName;
+
+        @Parameters(index = "1", description = "Configuration file path", arity = "0..1")
+        String filePath;
+
+        @Option(names = {"--overwrite"}, description = "Overwrite existing client configuration if it exists",
+                defaultValue = "false")
```

Comment:
Why false? I'd think invoking this command would indicate my desire to overwrite the existing config.

Reply:
We follow the same convention everywhere, by default don't overwrite and just warn.